sccp/figures/figureS2.py

```diff
@@ -0,0 +1,150 @@
+"""
+S2: Initial Attempt at Pf2 on the lupus data
+article: https://www.science.org/doi/10.1126/science.abf1970
+data: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE174188
+"""
+
+# GOAL: test Pf2 on lupus data, get visualizations for factor matrices
+
+# load functions/modules ----
+from .common import (
+    subplotLabel,
+    getSetup,
+    reorder_table
+)
+from ..parafac2 import parafac2_nd, Pf2X
+from ..imports.scRNA import tensorFy
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import anndata
+from .common import subplotLabel, getSetup
+import matplotlib.pyplot as plt
+import scipy.cluster.hierarchy as sch
+import tensorly as tl
+
+
+####################################################################################
+## LOAD IN DATA -- CAN BE MOVED TO imports/scRNA 
+####################################################################################
+
+def load_lupus_data(third_axis= "ind_cov", n_rand = 100):
+    """Import Lupus PBMC dataset."""
+    X = anndata.read_h5ad("/opt/andrew/lupus/lupus.h5ad")
+    
+    # reorder X so that all of the patients are in alphanumeric order. this is important
+    # so that we can steal cell typings at this point
+    obsV = X.obs_vector('ind_cov')
+    sgUnique, sgIndex = np.unique(obsV, return_inverse=True)
+
+    ann_data_objects = [X[sgIndex == sgi, :] for sgi in range(len(sgUnique))]
+
+    X = anndata.concat(ann_data_objects, axis = 0)
+
+    # select n_rand random patients; evenly split between lupus and SLE
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    # REMOVE THIS SECTION TO HAVE ALL INDIVIDUALS INCLUDED (and n_rand arg in function def)
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    np.random.seed(42)
+    SLE_patients = X.obs["ind_cov"][X.obs["SLE_status"] == "SLE"].to_numpy()
+    healthy_patients = X.obs["ind_cov"][X.obs["SLE_status"] == "Healthy"].to_numpy()
+    random_patients = np.random.choice(SLE_patients, int(n_rand/2), replace=False).tolist() + np.random.choice(healthy_patients, int(n_rand/2), replace=False).tolist()
+
+    X = X[X.obs['ind_cov'].isin(random_patients), :]
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+    # get cell types -> only stay true if order of cells doesn't change after this
+    cell_types = X.obs['cg_cov'].reset_index(drop=True)
+
+    # get color mapping for patients by SLE status
+
+    status = X.obs[["ind_cov","SLE_status"]].sort_values(by= "ind_cov").drop_duplicates("ind_cov")
+
+    lut = {'SLE': 'c', 'Healthy': 'm'}
+    row_colors = status['SLE_status'].map(lut)
+
+    assert np.all(np.isfinite(X.X.data)) # this should be true
+
+    return tensorFy(X, third_axis), cell_types, row_colors
+
+####################################################################################
+
+def plotLupusFactors(factors, data: Pf2X, row_colors, axs, reorder=tuple(), trim=tuple()):
```

Comment:
How is this function different from the existing plotFactors function? Ideally we would reuse that code.

Reply:
`PlotLupusFactors()` has a `row_colors` argument that sets the key to make a bar on the left hand side of the A matrix plot to visualize which patients were SLE/healthy. I can definitely add this as an optional argument to `PlotFactors()` that defaults to None (since each of the drugs are different in the Thomson lab dataset, we wouldn't necessarily need it for that); mostly I was just hesitant to change other files

Reply:
Feel free to add it to the master function!